csrc/rocm/attention.cu

```diff
@@ -63,23 +66,34 @@ typedef struct _B16x8 {
 } _B16x8;
 
 using _B8x8 = uint2;
+using _B8x4 = int32_t;  // used in builtins
+using bit8_t = uint8_t;
 
-////// Non temporal load stores ///////
+typedef struct _B8x16 {
+  _B8x8 xy[2];
+} _B8x16;
 
+////// Non temporal loads ///////
 template <typename T>
-__device__ __forceinline__ T load(T* addr) {
-  return addr[0];
+__device__ __forceinline__ T loadnt(T* addr) {
+  return __builtin_nontemporal_load(addr);
 }
 
-template <typename T>
-__device__ __forceinline__ void store(T value, T* addr) {
-  addr[0] = value;
+__device__ __forceinline__ _B16x8 load_ntmprl_16Byte(const _B16x8* addr) {
+  auto addr_alias = reinterpret_cast<const float*>(addr);
+  auto dat0 = loadnt(addr_alias);
+  auto dat1 = loadnt(addr_alias + 1);
+  auto dat2 = loadnt(addr_alias + 2);
+  auto dat3 = loadnt(addr_alias + 3);
+  auto res = make_float4(dat0, dat1, dat2, dat3);
+  return *reinterpret_cast<_B16x8*>(&res);
 }
+///////////////////////////////////
```

Comment:
Is this used anywhere? One general comment is that we should not try to add unused/untested/unexercised code to the codebase, so could you remove any other functions or types that aren't being used in this file?